R/bm-tpc-h.R

```diff
@@ -107,8 +107,17 @@ tpc_h <- Benchmark("tpc_h",
       all_equal_out <- waldo::compare(result, answer, tolerance = 0.01)
 
       if (length(all_equal_out) != 0) {
-        warning(paste0("\n", all_equal_out, "\n"))
-        stop("The answer does not match")
+        # DuckDB's data does not actually fit the spec for _address columns (the
+        # values in the answer(s) for those columns aren't found anywhere in the
+        # validation data or data generated with the official tool) so we try
+        # again with the duckdb specific answers
+        answer <- tpch_answer(scale_factor, query_id, datasource = "duckdb")
+        all_equal_out <- waldo::compare(result, answer, tolerance = 0.01)
+
+        if (length(all_equal_out) != 0) {
+          warning(paste0("\n", all_equal_out, "\n"))
+          stop("The answer does not match")
+        }
```

Comment:
It should only really be necessary temporarily to support both duckdb's data and the real thing from datalogistik. But this will help us not need to coordinate anything like cache clearing for now.